test/minikube/minikube_long_restore_test.go

```diff
@@ -891,3 +891,220 @@ func TestKubernetesRestoreDatabaseSeparateJournal(t *testing.T) {
+			"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.persistence.storageClass":     testlib.SNAPSHOTABLE_STORAGE_CLASS,
+			"database.name":                         sourceDb,
+		},
+	})
+
+	output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", admin0, "-c", "admin", "--", "bash", "-c",
+		fmt.Sprintf("echo \"CREATE TABLE testtbl (id INT); INSERT INTO testtbl (id) values (123);\" | nuosql %s --user dba --password secret", sourceDb))
+
+	require.NoError(t, err, output)
+
+	defer testlib.Teardown(testlib.TEARDOWN_SNAPSHOT)
+
+	smPod := fmt.Sprintf("sm-%s-nuodb-cluster0-%s-hotcopy-0", sourceDatabaseChartName, sourceDb)
+	achiveVolumeName := "archive-volume-" + smPod
+
+	noJournalNoBidSnapshotName := "noj-nobid-snapshot"
+	testlib.SnapshotVolume(t, namespaceName, achiveVolumeName, noJournalNoBidSnapshotName)
+
+	output, err = k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", smPod, "-c", "engine", "--", "bash", "-c",
+		fmt.Sprintf("echo \"{\\\"id\\\": \\\"123abc\\\"}\" > /var/opt/nuodb/archive/nuodb/%s/backup.json", sourceDb))
+	require.NoError(t, err, output)
+
+	noJournalBidSnapshotName := "noj-bid-snapshot"
+	testlib.SnapshotVolume(t, namespaceName, achiveVolumeName, noJournalBidSnapshotName)
+
+	output, err = k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", smPod, "-c", "engine", "--", "bash", "-c",
+		fmt.Sprintf("echo \"{\\\"id\\\": \\\"123abd\\\"}\" > /var/opt/nuodb/archive/nuodb/%s/backup.json", sourceDb))
+	require.NoError(t, err, output)
+
+	noJournalBadBidSnapshotName := "noj-badbid-snapshot"
+	testlib.SnapshotVolume(t, namespaceName, achiveVolumeName, noJournalBadBidSnapshotName)
+
+	output, err = k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", smPod, "-c", "engine", "--", "bash", "-c",
+		"mkdir /var/opt/nuodb/archive/nuodb/foo && touch /var/opt/nuodb/archive/nuodb/foo/info.json")
+	require.NoError(t, err, output)
+
+	noJournalExtraArchiveSnapshotName := "noj-dupe-arch-snapshot"
+	testlib.SnapshotVolume(t, namespaceName, achiveVolumeName, noJournalExtraArchiveSnapshotName)
+
+	helm.DeleteE(t, &options, sourceDatabaseChartName, true)
+
+	testDb("noj-nobid", noJournalNoBidSnapshotName, "", "", true)
+	testDb("noj-gdbid", noJournalBidSnapshotName, "", "123abc", true)
+	testDb("noj-ignbid", noJournalBadBidSnapshotName, "", "", true)
+
+	output = testDb("noj-misbid", noJournalNoBidSnapshotName, "", "123abc", false)
+	require.Contains(t, output, "Incorrect backup id in archive")
+
+	output = testDb("noj-badbid", noJournalBadBidSnapshotName, "", "123abc", false)
+	require.Contains(t, output, "Incorrect backup id in archive")
+
+	output = testDb("noj-2arch", noJournalExtraArchiveSnapshotName, "", "123abc", false)
+	require.Contains(t, output, "Did not find exactly 1 archive:")
+
+	sourceDb = "src-journ"
```

Comment:
This is the same test case as above but with a separate journal?

Reply:
Kind of. These are tests for recovering journals and mis-configured journals.